File: 2363-merge-similar-items/2363-merge-similar-items.js
/**
 * @param {number[][]} items1
 * @param {number[][]} items2
 * @return {number[][]}
 */
var mergeSimilarItems = function(items1, items2) {
    let obj = {};
    for(let [value, weight] of [...items1, ...items2]) {
        if(!obj[value]) obj[value] = weight;
        else obj[value] += weight;
    }
    return Object.entries(obj);
};

File: 2363-merge-similar-items/2363-merge-similar-items.py
class Solution:
    def mergeSimilarItems(self, items1: List[List[int]], items2: List[List[int]]) -> List[List[int]]:
        initial_dict = {}
        combine_list = [*items1, *items2]
        for (value, weight) in combine_list:
            if value in initial_dict:
                initial_dict[value] += weight
            else:
                initial_dict[value] = weight
            
        results = [(key, value) for key, value in initial_dict.items()]
        return sorted(results)